tsl/src/nodes/decompress_chunk/batch_queue_fifo.c

```diff
@@ -0,0 +1,83 @@
+/*
+ * This file and its contents are licensed under the Timescale License.
+ * Please see the included NOTICE for copyright information and
+ * LICENSE-TIMESCALE for a copy of the license.
+ */
+
+#include <postgres.h>
+
+#include "compression/compression.h"
+#include "nodes/decompress_chunk/batch_array.h"
+#include "nodes/decompress_chunk/batch_queue.h"
+#include "nodes/decompress_chunk/compressed_batch.h"
+
+static void
+batch_queue_fifo_free(BatchQueue *bq)
+{
+	batch_array_destroy(&bq->batch_array);
+}
+
+static bool
+batch_queue_fifo_needs_next_batch(BatchQueue *bq)
+{
+	return TupIsNull(batch_array_get_at(&bq->batch_array, 0)->decompressed_scan_slot);
+}
+
+static void
+batch_queue_fifo_pop(BatchQueue *bq, DecompressContext *dcontext)
+{
+	DecompressBatchState *batch_state = batch_array_get_at(&bq->batch_array, 0);
+	if (TupIsNull(batch_state->decompressed_scan_slot))
+	{
+		/* Allow this function to be called on the initial empty queue. */
+		return;
+	}
+
+	compressed_batch_advance(dcontext, batch_state);
+}
```

Comment:
Same problem here, these functions were inline and in a header for a reason -- to allow generating a streamlined implementation for the FIFO case.